spec-0007/index.md

```diff
@@ -0,0 +1,183 @@
+---
+title: "SPEC 7 — Seeding pseudo-random number generation"
+date: 2023-04-19
+author:
+  - "Stéfan van der Walt <stefanv@berkeley.edu>"
+  - "Sebastian Berg <sebastianb@nvidia.com>"
+  - Other participants in the discussion <not.yet@named.org>"
+discussion: https://github.com/scipy/scipy/issues/14322
+endorsed-by:
+---
+
+## Description
+
+<!--
+Briefly and clearly describe the proposal.
+Explain the general need and the advantages of this specific proposal.
+If relevant, include examples of how the new functionality would be used,
+intended use-cases, and pseudo-code illustrating its use.
+-->
+
+There is disparity in the APIs libraries provide to seed random number generation.
+This SPEC suggests a single, pragmatic API for the ecosystem, taking into account technical and historical factors.
+Adopting such a uniform API will simplify the user experience, especially for those who rely on multiple projects.
+
+Specifically, we recommend to:
+
+- Deprecate the use of `RandomState` and `np.random.seed`.
+- Standardize usage and interpretation of an `rng` keyword for seeding.
```

Comment:
I think we can split this into two via something like

```suggestion
- Pass the default or seeded `numpy.random.Generator` instances to functions.
- Standardize the `rng` acronym for the keyword in function signatures.
```

so that the context is separated from the keyword standardization and the actual deprecation. Please adjust the sentences as you see fit.

Reply:
Thanks for the feedback, @ilayn. Currently, the recommendation is not to follow any specific technical path, but to (a) avoid global state and (b) use a standard mechanism for sending generators around.

Reading your suggestion, I think it could be added as a second point, i.e., write down the principles, followed by how they should be enacted. E.g.:

```
We recommend:

- Standardizing the usage and interpretation of an `rng` keyword for seeding, and
- avoiding the use of global state and legacy bitstream generators.

We suggest implementing these principles by:

- Passing the `rng` argument to `numpy.random.default_rng` to instantiate a `Generator`, and
- deprecating  the use of `RandomState` and `np.seed`.
```

Thoughts?

Reply:
Ah yes just wanted to make sure that implementation does not get mixed up with the SPEC goal. All good from my side.